### PR TITLE
*: use the capabilities API to drive center selector screen

### PR DESCRIPTION
**Pull request by stevekuznetsov**
The new capabilities API lets us know which centers are registered in the NAC system and for which cepabilities they've signed up. This PR adds the data fetching as well as using this data to drive the center selection screen(s). A follow-up will start to use the capabilities enumerated to determine when we show which parts of the app.

IMG_0108

### CODE REVIEW

File: clientContext.ts [hunk 1]
```diff
@@ -2,18 +2,21 @@ import React, {Context} from 'react';
...
 export interface ClientProps {
   nationalAvalancheCenterHost: string;
+  nationalAvalancheCenterWordpressHost: string;
   snowboundHost: string;
   nwacHost: string;
 }
 
 export const productionHosts = {
   nationalAvalancheCenterHost: 'https://api.avalanche.org',
+  nationalAvalancheCenterWordpressHost: 'https://devavycenters.wpengine.com', // TODO(skuznets): what's the production version here?
```
Comment (stevekuznetsov) [inline clientContext.ts:12]: https://avycenters.wpengine.com

This is likely it, but it does not (yet) host the API. I'll update this when Chris gets back to us.

File: components/avalancheCenterList.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  if (centers === AvalancheCenters.SupportedCenters) {
+    whichCenters = metadata.filter(center => supportedAvalancheCenters.includes(center.id as AvalancheCenterID));
+  } else {
+    whichCenters = metadata;
+  }
+
+  return whichCenters.map(center => ({
+    center: center.id as AvalancheCenterID,
+    name: center.name,
+    title: `The ${center.name} (${center.id}) produces avalanche forecasts for ${center.zones.length} zones out of ${center.city}, ${center.state}.`,
```
[the remark this answers is not in the document]
Comment (floatplane) [reply components/avalancheCenterList.ts:40]: This approach also doesn't pluralize correctly

image

**Review by floatplane — COMMENTED** [review]
Some thoughts below...also, there was discussion in the design Figma about having some more descriptive text next to the name of the avalanche centers, which is why I added it in the first place. 

Does this change the appearance of the regular center picker, or is it only changing the debug one?

File: components/avalancheCenterList.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (stevekuznetsov) [reply components/avalancheCenterList.ts:40]: What should we say that is not going to be repetitive? Once we have a design I'm happy to update it. IMHO I would just have list element cards that have the ID abbreviation, full name and city + state. If we want to have an optional config that plugs in per-center blurbs, we can do that as well.

File: components/content/AvalancheCenterList.tsx [hunk 3]
```diff
@@ -21,6 +21,7 @@ const AvalancheCenterListItem: React.FC<AvalancheCenterListItemProps> = ({data,
         <HStack space={8} flexShrink={1} alignItems="flex-start">
           <AvalancheCenterLogo style={{height: 24, width: 24, resizeMode: 'contain', flex: 0, flexGrow: 0}} avalancheCenterId={data.center} />
           <VStack space={2} flexShrink={1}>
+            <BodyBlack>{data.name}</BodyBlack>
```
[the remark this answers is not in the document]
Comment (stevekuznetsov) [reply components/content/AvalancheCenterList.tsx:24]: I was following the wise words of the previous iteration

> Didn't spend a lot of time making the debug version of this pretty - it just scales the icons way way down and uses the avalanche center IDs. But the code is almost entirely shared now.

[the remark this answers is not in the document]
Comment (stevekuznetsov) [reply components/content/AvalancheCenterList.tsx:24]: In all seriousness - are we expecting a full design for this? If so, can we defer fiddling with the details until that's done?

**Comment by stevekuznetsov** [comment]
Updated the visuals to be nicer. This updates the way both of the screens are rendered but your distinction between which set of centers to render where remains the same

File: components/avalancheCenterList.ts [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+  for (const center of ids) {
+    const idResult = avalancheCenterIDSchema.safeParse(center);
+    if (idResult.success) {
+      knownCenters.push(idResult.data);
+    }
+  }
+  return knownCenters;
+};
+
+const blurbs: Record<AvalancheCenterID, string> = {
```
Comment (stevekuznetsov) [inline components/avalancheCenterList.ts:27]: I imagine there's some way to indicate on the type level that this record has *some* but not all entries, but as written `tsc` yelled at me and made me add in every possible key.

**Pull request merged**